packages/rest-crud/src/repository-builder.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/rest-crud
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  DefaultCrudRepository,
+  Entity,
+  EntityCrudRepository,
+  juggler,
+} from '@loopback/repository';
+import * as assert from 'assert';
+
+/**
+ * Create (define) a repository class for the given model.
+ *
+ * Example usage:
+ *
+ * ```ts
+ * const ProductRepository = defineCrudRepositoryClass(Product);
+ * ```
+ *
+ * @param modelCtor A model class, e.g. `Product`.
+ */
+export function defineCrudRepositoryClass<
+  T extends Entity,
+  IdType,
+  Relations extends object = {}
+>(
+  entityClass: typeof Entity & {prototype: T},
+): RepositoryClass<T, IdType, Relations> {
+  const repoName = entityClass.name + 'Repository';
+  const defineNamedRepo = new Function(
+    'EntityCtor',
+    'BaseRepository',
+    `return class ${repoName} extends BaseRepository {
+      constructor(dataSource) {
```

Comment:
I would like to make it possible to set up `@inject` here too.

Reply:
What should it be able to `@inject`?

Reply:
Injection of the `datasource`, for example:

```ts
class ${repoName} extends BaseRepository {
      constructor(@inject(${dataSourceBinding}) dataSource: DataSource) {}
}